trie/stacktrie.go

```diff
@@ -211,15 +205,14 @@ func (st *StackTrie) Reset() {
 		st.children[i] = nil
 	}
 	st.nodeType = emptyNode
-	st.keyOffset = 0
 }
 
 // Helper function that, given a full key, determines the index
 // at which the chunk pointed by st.keyOffset is different from
 // the same chunk in the full key.
 func (st *StackTrie) getDiffIndex(key []byte) int {
 	diffindex := 0
-	for ; diffindex < len(st.key) && st.key[diffindex] == key[st.keyOffset+diffindex]; diffindex++ {
+	for ; diffindex < len(st.key) && st.key[diffindex] == key[diffindex]; diffindex++ {
```

Comment:
I _think_ this method woud become clearer by writing it like this
```go
func (st *StackTrie) getDiffIndex(key []byte) int {
	for idx, k := range st.key{
		if k != key[idx]{
			return idx
		}
	}
	return len(key)
}
```

Reply:
Sure, will do.